File: 344.Reverse String/solution.py
class Solution(object):
    def recursion(self, s, i, j):
        if(i>=j): return
        s[j],s[i] = s[i],s[j]
        self.recursion(s,i+1,j-1)
        
    def reverseString(self, s):
        """
        :type s: List[str]
        :rtype: None Do not return anything, modify s in-place instead.
        """
        i = 0
        j = len(s)-1
        self.recursion(s,i,j)
        

File: 344.Reverse String/solution.java
class Solution {
    public void reverseString(char[] s) {
        int n = s.length;
        int h = n-1;
        int l = 0;
        while(l<h){
            // int a = (int)s[l];
            // int b = (int)s[h];
            
            // a = a+b;
            // b = a-b;
            // a = a-b;
            s[l] = (char)((int)s[l]+(int)s[h]);
            s[h] = (char)((int)s[l]-(int)s[h]);
            s[l] = (char)((int)s[l]-(int)s[h]);
            // s[l] = (char)a;
            // s[h] = (char)b;
            
            l++;
            h--;
        }
    }
}